application-admintools-default/src/main/java/com/xwiki/admintools/internal/usage/SpamPagesProvider.java

```diff
@@ -44,78 +43,46 @@
-            }
-        });
-        applyDocumentsSort(spammedDocuments, sortColumn, order);
-        return spammedDocuments;
-    }
-
-    /**
-     * Get the references of documents in wiki with comments above a given limit.
-     *
-     * @param maxComments maximum number of comments below which the document is ignored.
-     * @param searchedDocument document hint to be searched.
-     * @param wikiId the wiki for which the data will be retrieved.
-     * @return a {@link List} with the {@link DocumentReference} of the documents with comments above a given limit.
+     * @return a {@link SolrDocumentList} with the needed fields set.
      * @throws QueryException if there are any exceptions while running the queries for data retrieval.
      */
-    public List<DocumentReference> getCommentsForWiki(long maxComments, String searchedDocument, String wikiId)
-        throws QueryException
+    public SolrDocumentList getDocumentsOverGivenNumberOfComments(long maxComments, Map<String, String> filters,
+        String order) throws Exception
     {
-        String searchString;
-        if (searchedDocument == null || searchedDocument.isEmpty()) {
-            searchString = "%";
-        } else {
-            searchString = String.format("%%%s%%", searchedDocument);
+        String searchedDocument = filters.get("docName");
+        String queryStatement = "*";
+        if (searchedDocument != null && !searchedDocument.isEmpty()) {
+            queryStatement = String.format("title:%s", solrUtils.toCompleteFilterQueryString(searchedDocument));
+        }
+        List<String> filterStatements = new ArrayList<>();
+        filterStatements.add("type:DOCUMENT");
+        filterStatements.add(String.format("AdminTools.NumberOfComments_sortInt:[%d TO *]", maxComments));
+
+        Query query = this.secureQueryManager.createQuery(queryStatement, "solr");
+        if (query instanceof SecureQuery) {
+            ((SecureQuery) query).checkCurrentAuthor(true);
+            ((SecureQuery) query).checkCurrentUser(true);
         }
-        return this.queryManager.createQuery("select obj.name from XWikiDocument as doc, BaseObject as obj "
-                + "where doc.fullName = obj.name and obj.className = 'XWiki.XWikiComments' "
-                + "and lower(doc.title) like lower(:searchString) "
-                + "group by obj.name having count(*) > :maxComments order by count(*) desc", Query.HQL).setWiki(wikiId)
-            .bindValue("maxComments", maxComments).bindValue("searchString", searchString)
-            .addFilter(currentLanguageFilter).addFilter(hiddenDocumentFilter).addFilter(documentFilter)
-            .addFilter(viewableFilter).execute();
+
+        query.bindValue("fl", "title_, reference, wiki, AdminTools.NumberOfComments_sortInt, links, name, spaces");
+        query.bindValue("fq", filterStatements);
+        query.bindValue("sort", String.format("AdminTools.NumberOfComments_sortInt %s", order));
```

Comment:
Is there any validation for order to be one of the supported values? It's not critical for security I think, it just seems like a good idea to validate inputs.